addons/auto-follow-topics/module.js

```diff
@@ -0,0 +1,32 @@
+// This module should run after page load - only use this with `runAtComplete:true`
+const getTopics = () => JSON.parse(sessionStorage.getItem("auto-follow-topics") || "[]");
+const setTopics = (topics) => sessionStorage.setItem("auto-follow-topics", JSON.stringify(topics));
+const postError = () => Boolean(document.querySelector(".errorlist"));
+const topicID = location.href.split("/")[5];
+
+document.querySelector("[name=AddPostForm]")?.addEventListener("click", (event) => {
+  const ids = getTopics();
+  ids.push(topicID);
+  setTopics(ids);
+  onPostCallbacks.forEach((callback) => callback());
+});
+
+const topics = getTopics();
+let success = false;
+const hasTopic = topics.includes(topicID);
+if (hasTopic && !postError()) {
+  success = true;
+  setTopics(topics.filter((topic) => topic !== topicID));
+}
+
+const onPostCallbacks = [];
+export const onPost = (callback) => {
+  onPostCallbacks.push(callback);
+};
+
+export const onSuccessfulPost = (callback) => {
+  if (hasTopic && success) callback(topicID);
+};
+export const onNonSuccessfulPost = (callback) => {
+  if (hasTopic && !success) callback(topicID);
```

Comment:
This still isn't convincing to me. Shouldn't this check for `postError()`?

Reply:
`postError()` is called when setting the `success` variable.

Reply:
`success` being true does guarantee there wasn't any post errors. However, `success` being false tells us nothing about whether there was a post error or not. Again, how do we want to define a "non successful post"?

Reply:
The variable `success` is true, if and only if, `hasTopic && !postError()` is true
That is, `!success` is true if either `hasTopic` if false, `postError()` is true, or both at the same time

To see if there's a non-successful post, you evaluate `hasTopic && !success`.
But as said above, this is equivalent to `hasTopic && (!hasTopic || postError())`
Of course, `hasTopic` cannot be true and false at the same time, so the condition you end up checking is `hasTopic && postError()`

So I guess your code is equivalent to what I would expect (I would need to check again), but I found it quite confusing!

Reply:
Should I just leave out the `success` variable in favor of calling `postError` directly?

Reply:
```suggestion
export const onNonSuccessfulPost = (callback) => {
  if (postError()) callback(topicID);
```

I guess this would work as you expect?